docs/data/date-pickers/experimentation/CustomField.tsx

```diff
@@ -0,0 +1,73 @@
+import * as React from 'react';
+import dayjs, { Dayjs } from 'dayjs';
+import { DemoContainer } from '@mui/x-date-pickers/internals/demo';
+import { AdapterDayjs } from '@mui/x-date-pickers/AdapterDayjs';
+import { LocalizationProvider } from '@mui/x-date-pickers/LocalizationProvider';
+import { DatePicker } from '@mui/x-date-pickers/DatePicker';
+import TextField from '@mui/material/TextField';
+import {
+  BaseSingleInputFieldProps,
+  DateValidationError,
+  FieldSection,
+} from '@mui/x-date-pickers/models';
+import { useValidation, validateDate } from '@mui/x-date-pickers/validation';
+import { UseDateFieldProps } from '@mui/x-date-pickers/DateField';
+import {
+  BaseDateValidationProps,
+  DefaultizedProps,
+  splitFieldInternalAndForwardedProps,
+} from '@mui/x-date-pickers/internals';
+
+interface ReadOnlyDateFieldProps
+  extends DefaultizedProps<
+      UseDateFieldProps<Dayjs, false>,
+      keyof BaseDateValidationProps<Dayjs>
+    >,
+    BaseSingleInputFieldProps<
+      Dayjs | null,
+      Dayjs,
+      FieldSection,
+      false,
+      DateValidationError
+    > {}
+
+function ReadOnlyField(props: ReadOnlyDateFieldProps) {
+  const { internalProps, forwardedProps } = splitFieldInternalAndForwardedProps<
```

Comment:
`splitFieldInternalAndForwardedProps` is private for now
It would probably worth making it public and documenting it properly.

Without it it's not trivial to know which props to forward to the DOM or not.